# backend/taskorchestrator/tasks.py

from celery import shared_task
from taskorchestrator.utils.dispatcher import run_task
from taskorchestrator.utils.logger_config import logger  # ğŸ”¥ Merkezi log kullanÄ±mÄ±


@shared_task(bind=True, name="taskorchestrator.tasks.run_scheduled_task")
def run_scheduled_task(self, task_id: int):
    """
    Celery tarafÄ±ndan tetiklenen ana gÃ¶rev fonksiyonu.
    Verilen ScheduledTask ID'sine gÃ¶re ilgili fonksiyonu Ã§alÄ±ÅŸtÄ±rÄ±r.
    Sadece hata durumlarÄ± loglanÄ±r.
    """
    try:
        run_task(task_id)
    except Exception as e:
        logger.exception(f"ğŸ”¥ Celery run_scheduled_task hatasÄ±: {str(e)}")
